script/agencies_verify.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import requests
+import sys
+import urllib
+import yaml
+
+sys.path.append("services/gtfs-rt-archive/src/gtfs_rt_archive/")
+
+from mapperfns import map_agencies_urls, map_headers  # noqa
+
+USAGE = """
+Usage:
+python script/agencies_verify.py agencies.yml headers.yml [DIFF]
+
+Arguments:
+agencies.yml - Yaml file containing agencies to download
+headers.yml - Yaml file mapping headers onto urls in agencies.yml
+DIFF - optional git diff between two hashes (will check all urls if not provided)
+"""
+
+
+def main():
+    logger = logging.getLogger("gtfs-rt-archive")
+    successes = []
+    fails = []
+    changed_domains = None
+
+    with open(sys.argv[1], "r") as f:
+        agencies = map_agencies_urls(logger, yaml.load(f, Loader=yaml.SafeLoader))
+    with open(sys.argv[2], "r") as f:
+        headers = dict(list(map_headers(logger, yaml.load(f, Loader=yaml.SafeLoader))))
+    if len(sys.argv) > 3:
+        with open(sys.argv[3], "r") as f:
+            lines = f.readlines()
+            lines = [line for line in lines if line.startswith("+") and "http" in line]
+            changed_urls = ["http" + line.strip().split("http")[-1] for line in lines]
+            changed_domains = [
+                urllib.parse.urlparse(url).netloc for url in changed_urls
```

Comment:
This checking of domains is problematic. If there is ever a change to a Swiftly URL or a 511 URL, that will result in all URLs on that domain being matched and it will trigger the checking of all the Swiftly and 511 URLs and not just those that changed. I think it would be best to not compile a list of domains and instead focus on just the exact URLs that get stored into the `changed_urls` variable.

Reply:
Resolved in 34b6287.

I used changed_domains the first time because the AGENCIES_DIFF file is between versions of `airflow/data/agencies.yml` which doesn't have secret keys in them. The script that replaces the secret keys (script/yml_convert.py) wasn't set up to work with non-yaml files. In my most recent I gave the yml_convert script an `--unsafe` option, so using changed_urls should work now.